index.html

```diff
@@ -10198,127 +10215,229 @@ <h2>User prompts</h2>
- as a non-normative reference:
+<p>The <dfn>current user prompt</dfn> is the <a>user prompt</a> which
+has invoked the <a data-cite=html>pause</a> algorithm, if any, or null otherwise.
 
-<table class=simple>
- <tr>
-  <th>Definition
-  <th>Dialog
-  <th>Interactions
- </tr>
+<p>To <dfn data-lt="dismissed|dismisses">dismiss</dfn> the <a>current
+ user prompt</a>, act as if the user clicked the <b>Cancel</b> button
+ prompt on that prompt, if present, or otherwise <a>accept</a> the
+ prompt.
 
- <tr>
-  <td><dfn>Alert</dfn>
-  <td><a>window.<code>alert</code></a>
-  <td>
-   <a>Accept Alert</a><br>
-   <a>Dismiss Alert</a><br>
-   <a>Get Alert Text</a>
- </tr>
+<p>To <dfn data-lt="accepting|accepted|accepts">accept</dfn>
+ the <a>current user prompt</a>, act if the user clicked
+ the <b>OK</b> button on that prompt.
 
- <tr>
-  <td><dfn>Confirm</dfn>
-  <td><a>window.<code>confirm</code></a>
-  <td>
-   <a>Dismiss Alert</a><br>
-   <a>Accept Alert</a><br>
-   <a>Get Alert Text</a>
- </tr>
+<h3>User Prompt Handler</h3>
 
- <tr>
-  <td><dfn>Prompt</dfn>
-  <td><a>window.<code>prompt</code></a>
-  <td>
-   <a>Dismiss Alert</a><br>
-   <a>Accept Alert</a><br>
-   <a>Get Alert Text</a><br>
-   <a>Send Alert Text</a>
- </tr>
-</table>
+<p>A <a>remote end</a> has a <dfn>user prompt handler</dfn> which
+defines how a WebDriver session will react when a user prompt is
+displayed. It is either null or a <a data-cite=infra>map</a> between strings
+and <a>prompt handler configuration</a> values. Initially it is null.
+
+<p>A <dfn>prompt handler configuration</dfn> is a <a>struct</a> with two
+items; a <dfn data-dfn-for="prompt handler configuration">handler</dfn>, which is a
+string, and a <dfn data-dfn-for="prompt handler configuration">notify</dfn>,
+which is a boolean.
 
-<p>The <dfn>current user prompt</dfn> is said to be the active <a>user prompt</a>,
- which can be one of the entries on the <a>table of simple dialogs</a>.
+<p>To <dfn>serialize a prompt handler configuration</dfn>
+given <var>configuration</var>:
 
-<p>To <dfn data-lt="dismissed|dismisses">dismiss</dfn>
- the <a>current user prompt</a>,
- do so as if the user would click the <b>Cancel</b> or <b>OK</b> button,
- whichever is present, in that order.
+<ol class=algorithm>
 
-<p>To <dfn data-lt="accepting|accepted|accepts">accept</dfn> the <a>current user prompt</a>,
- do so as if the user would click the <b>OK</b> button.
+ <li><p>Let <var>serialized</var>
+ be <var>configuration</var>&apos;s [=prompt handler
+ configuration/handler=].
 
-<p>
-The <dfn>user prompt handler</dfn> defines what action
-the <a>remote end</a> must take when a <a>user prompt</a> is encountered.
-This is defined by the <a>unhandled prompt behavior</a> capability.
-The following <dfn>known prompt handling approaches table</dfn>
-lists the keywords and states for the attribute:
+ <li><p>If «"<code>dismiss</code>", "<code>accept</code>"»
+ [=list/contains=] <var>serialized</var>,
```

Comment:
Maybe the following? That way we don't have to handle a list and check for the item to be contained.

```suggestion
 <li><p>If <var>serialized</var> is not equal to "<code>ignore</code>",
```

Reply:
Good argument. Then it's fine for me.